cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/SimulationState.java

```diff
@@ -12,19 +12,108 @@
     private final List<CommandInvocationNode> nodes;
     private int position;
     private final Map<String, Set<CommandInvocationNode>> variables;
-    private final Map<String, LogicalBlock> subroutines = new HashMap<>();
+    private final Set<CommandInvocationNode> suspiciousPoints;
+    private final Map<String, AbstractSimulator.CommandSimulator> dynamicSimulators = new HashMap<>();
 
-    SimulationState(List<CommandInvocationNode> nodes, int position) {
-        this(nodes, position, new HashMap<>());
+    SimulationState(List<CommandInvocationNode> nodes, int position, Set<CommandInvocationNode> suspiciousPoints) {
+        this(nodes, position, suspiciousPoints, new HashMap<>());
     }
 
-    SimulationState(List<CommandInvocationNode> nodes, int position, Map<String, Set<CommandInvocationNode>> variables) {
+    SimulationState(List<CommandInvocationNode> nodes, int position, Set<CommandInvocationNode> suspiciousPoints, Map<String, Set<CommandInvocationNode>> variables) {
         this.nodes = nodes;
         this.position = position;
         this.variables = variables;
+        this.suspiciousPoints = suspiciousPoints;
     }
 
-    void simulate(Set<Node> suspiciousPoints, Node node) {
+    /** todo think of better name */
+    String getValue(Node argumentNode) {
+        StringBuilder sb = new StringBuilder();
+
+        argumentNode.visitAll(new NodeVisitorAdapter() {
+            @Override
+            public void accept(ExpressionNode node) {
+                StringBuilder exprBuilder = new StringBuilder();
+
+                if (node.getKey() != null) {
+                    exprBuilder.append(node.getKey());
+                }
+
+                node.getNested().stream().forEach(n -> exprBuilder.append(getValue(n)));
+
+                String expr = exprBuilder.toString();
+                processUsage(expr);
+
+                sb.append('*').append(expr);
+            }
+
+            @Override
+            public void accept(ConstantNode node) {
+                String value = node.getValue();
+                if (value.endsWith("\n")) {
+                    value = value.replace("\n", ""); // todo this hack should be replaced by fix in parser!
+                }
+                sb.append(value);
+            }
+        });
+
+        return sb.toString();
+    }
+
+    void processUsage(ArgumentNode argumentNode) {
+        getValue(argumentNode);
+
+        processUsage(getArgument(argumentNode));
+    }
+
+    protected void processUnknownUsage(String variable) {
+
+    }
+
+    void processUsage(String variable) {
+        Set<CommandInvocationNode> nodes = variables.get(variable);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "nodes" which hides the field declared at line 12. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)